juntagrico/templatetags/juntagrico/depot_extras.py

```diff
@@ -24,27 +26,45 @@ def count(entity):
 def count_units(subs_or_types):
     if not subs_or_types:
         return 0
+    current_week_number = datetime.now().isocalendar()[1] - 1
     if isinstance(subs_or_types, Subscription):
         # case 1: single subscription object is passed
-        units = subs_or_types.active_parts.aggregate(units=Sum('type__size__units'))
+        units = (subs_or_types.active_parts
+                              .annotate(week_mod=ExpressionWrapper((current_week_number + F('type__offset')) % (F('type__interval')),
+                                                                   output_field=PositiveIntegerField()))
+                              .filter(week_mod=0)
+                              .aggregate(units=Sum('type__size__units')))
     elif isinstance(subs_or_types, QuerySet):
         if subs_or_types.model is Subscription:
             # case 2: sum each unit of each subscription type
-            units = {'units': str(sum([float(sub.active_parts.aggregate(units=Sum('type__size__units'))['units'] or 0) for sub in subs_or_types.all()]))}
+            units = {'units': str(sum([float(sub.active_parts
+                                             .annotate(week_mod=ExpressionWrapper((current_week_number + F('type__offset')) % (F('type__interval')),
+                                                                                  output_field=PositiveIntegerField()))
+                                             .filter(week_mod=0)
+                                             .aggregate(units=Sum('type__size__units'))['units'] or 0) for sub in subs_or_types.all()]))}
         elif subs_or_types.model is SubscriptionType:
             # case 3: queryset of types is passed
-            units = subs_or_types.aggregate(units=Sum('size__units'))
+            units = (subs_or_types.annotate(week_mod=ExpressionWrapper((current_week_number + F('offset')) % (F('frequency')),
+                                                                       output_field=PositiveIntegerField()))
+                                  .filter(week_mod=0)
+                                  .aggregate(units=Sum('size__units')))
     return float(units['units'] or 0)
 
 
 @register.filter
 def get_types_by_size(subscriptions, size):
+    current_week_number = datetime.now().isocalendar()[1]-1
```

Comment:
Taking the current week will not work properly if e.g. the depotlist is generated on Saturday for a delivery on next week Monday.
This is actually part of a bigger issue, as also `q_isactive()` below, that was there before, uses the current date instead of the actual delivery date. I'll analyse the problem in more detail and will open a separate issue or PR.

Reply:
Should we merge this as is or should I find a solution for it in this PR?

Reply:
I have an idea how to solve this and will create a PR for it. Then this PR can be adjusted.

Reply:
Here is the PR to fix date issue. #627
This fix will just allow to change the reference date, such that organisations that generate the list on a saturday and deliver on a monday can tell the list to look for active parts on the next monday.

I also applied the custom queryset approach. That way you can add your week filter to `SubscriptionPartQuerySet` by overriding the `active_on` method there.